### PR TITLE
codeintel: soft-fail and alert on codeintel-qa failure

### PR DESCRIPTION
**Pull request by Strum355**
does what it says on the tin

## Test plan

n/a, is doing test stuff

### CODE REVIEW

File: enterprise/dev/ci/internal/ci/operations.go [hunk 1]
```diff
@@ -557,6 +557,13 @@ func triggerReleaseBranchHealthchecks(minimumUpgradeableVersion string) operatio
 func codeIntelQA(candidateTag string) operations.Operation {
 	return func(p *bk.Pipeline) {
 		p.AddStep(":docker::brain: Code Intel QA",
+			bk.SlackStepNotify(&bk.SlackStepNotifyConfigPayload{
```
Comment (jhchabran) [inline enterprise/dev/ci/internal/ci/operations.go:560]: Ah damn! I should have told you, I submitted the same exact code, just didn't open the PR before going to have lunch 😅

File: enterprise/dev/ci/internal/ci/operations.go [hunk 2]
```diff
@@ -557,6 +557,13 @@ func triggerReleaseBranchHealthchecks(minimumUpgradeableVersion string) operatio
...
+				Message:     ":alert: :noemi-handwriting: Code Intel QA Flake detected <@U018WMLUL3F>",
```
Comment (jhchabran) [inline enterprise/dev/ci/internal/ci/operations.go:561]:
```suggestion
				Message:     ":alert: :noemi-handwriting: Code Intel QA Flake detected <@code-intel-support>",
```

The plugin handles that for us.

File: enterprise/dev/ci/internal/ci/operations.go [hunk 3]
```diff
@@ -557,6 +557,13 @@ func triggerReleaseBranchHealthchecks(minimumUpgradeableVersion string) operatio
...
+				ChannelName: "code-intel",
+				Conditions: bk.SlackStepNotifyPayloadConditions{
+					Failed: true,
```
Comment (jhchabran) [inline enterprise/dev/ci/internal/ci/operations.go:564]:
```suggestion
					Failed: true,
					Branches: []string{"main"},
```

Comment (jhchabran) [reply enterprise/dev/ci/internal/ci/operations.go:564]: Unless you want to get notification for all branches if you ever run this step elsewhere than `main`

**Review by jhchabran — CHANGES_REQUESTED** [review]
Also, need to merge the doc PR first, as the default value for the slack token is there.

File: enterprise/dev/ci/internal/ci/operations.go (hunk 2 above)
Comment (Strum355) [reply enterprise/dev/ci/internal/ci/operations.go:561]: This pings me directly, not code-intel-support. Id rather I get paged directly :slightly_smiling_face:

File: enterprise/dev/ci/internal/ci/operations.go (hunk 3 above)
Comment (Strum355) [reply enterprise/dev/ci/internal/ci/operations.go:564]: Yea if it happens in any dry-run branches or anywhere else, I want to know, so we dont miss any

File: enterprise/dev/ci/internal/ci/operations.go (hunk 1 above)
Comment (Strum355) [reply enterprise/dev/ci/internal/ci/operations.go:560]: :sonic:

File: enterprise/dev/ci/internal/ci/operations.go (hunk 2 above)
Comment (jhchabran) [reply enterprise/dev/ci/internal/ci/operations.go:561]: Ack, let's put your nick there then 😊

File: enterprise/dev/ci/internal/ci/operations.go (hunk 3 above)
Comment (jhchabran) [reply enterprise/dev/ci/internal/ci/operations.go:564]: Ack, perfect 👍

File: enterprise/dev/ci/internal/ci/operations.go (hunk 2 above)
Comment (Strum355) [reply enterprise/dev/ci/internal/ci/operations.go:561]: Does that work? Not sure how/if it works given theres a space in it. ID seems to the most likely to be correct :thinking:

Comment (jhchabran) [reply enterprise/dev/ci/internal/ci/operations.go:561]: https://sourcegraph.slack.com/archives/C02L31KTHHT/p1657714533415209 Yeah it does! 

```
			bk.SlackStepNotify(&bk.SlackStepNotifyConfigPayload{
				Message:              "CodeIntelQA :brain:,  checking if spaces work cc <@Noah S-C>",
```

You're right though, it should be possible to pass slack uids instead. I'll patch the plugins code along with the rest after this PR. 

For the time being, it doesn't work with uids, so for now 

```suggestion
				Message:     ":alert: :noemi-handwriting: Code Intel QA Flake detected <@Noah S-C>",
```
is the only option.

**Review by mrnugget — APPROVED** [review]

**Review by jhchabran — APPROVED** [review]

**Pull request merged**